### ISSUE TITLE
Is there a way to extract positions/postings information from a search result?

### ISSUE THREAD

**Issue by ferruccio** [quickwit-inc/tantivy#2271]
I am trying to highlight search results within their original documents for presentation. In order to do that I need to get the raw position information for each term involved in the search, but have been unable to find a way to do that. I'm not even sure that information is kept around when a search is completed. The document text is indexed with STORED & POSITION options.

**Comment by adamreichold** [comment]
Maybe the code from [`SnippetGenerator`](https://docs.rs/tantivy/latest/tantivy/struct.SnippetGenerator.html) is helpful?

**Comment by ferruccio** [comment]
Yes, I did look at the SnippetGenerator source and it was helpful. I was kind of hoping there was an API I had missed which generates a list of positions given a query and a document id as parameters. But, I think I see a path to building that kind of functionality by using an InvertedIndexReader to extract the position information for each search term within that document and then merging those results.
